x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/inference_service_form_fields.tsx

```diff
@@ -299,19 +319,52 @@ export const InferenceServiceFormFields: React.FC<InferenceServicesProps> = ({
     ]
   );
 
+  const getUpdatedProviders = useCallback(
+    (filterBySolution?: SolutionView) => {
+      if (providers) {
+        const filteredProviders = filterBySolution
+          ? providers.filter((provider) => {
+              const providerSolutions =
+                SERVICE_PROVIDERS[provider.service as ServiceProviderKeys]?.solutions ?? [];
+              return (
+                !solutionKeys[filterBySolution] ||
+                (solutionKeys[filterBySolution] !== undefined &&
+                  providerSolutions.includes(solutionKeys[filterBySolution] as ProviderSolution))
+              );
+            })
```

Comment:
Suggestion to extract this block of logic within filter out to its own helper function. I think it would be more readable (something like `providers.filter(isProviderForSolution)`)  and testable.